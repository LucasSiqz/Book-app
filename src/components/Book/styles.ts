import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    cursor: pointer;
    transition: opacity ${theme.transition.default};

    &:hover {
      opacity: 0.8;
    }

    strong {
      color: rgba(49, 49, 49, 0.8);
      font-size: ${theme.font.sizes.xsmall};
    }

    strong:nth-child(3) {
      font-size: 1rem;
      margin-top: 0.5rem;
    }
  `}
`

export const Image = styled.img`
  ${({ theme }) => css`
    border-radius: ${theme.border.radius};
    width: 10rem;
    height: 15rem;
    filter: drop-shadow(0px 2px 4px rgba(229, 229, 229, 0.5));
    margin-bottom: 1rem;
  `}
`
