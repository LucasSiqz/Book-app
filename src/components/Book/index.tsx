import { useRouter } from 'next/router'

import * as S from './styles'

export type BookProps = {
  id: string
  title: string
  author: string
  image?: string
}

const Book = ({ id, title, author, image }: BookProps) => {
  const { push } = useRouter()

  return (
    <S.Wrapper data-testid={id} onClick={() => push(`details/${id}`)}>
      <S.Image src={image || 'img/cover.png'} alt={title} />
      <strong>{title}</strong>
      <strong>{`by ${author}`}</strong>
    </S.Wrapper>
  )
}
export default Book
