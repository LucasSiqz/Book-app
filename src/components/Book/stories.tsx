import { Story, Meta } from '@storybook/react/types-6-0'
import Book, { BookProps } from '.'

export default {
  title: 'Book',
  component: Book,
  args: {
    title: 'HOOKED',
    author: 'Nir Eyal',
    image: 'img/Book.png'
  }
} as Meta

export const Default: Story<BookProps> = (args) => <Book {...args} />
