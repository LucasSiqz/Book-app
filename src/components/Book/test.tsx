import userEvent from '@testing-library/user-event'
import { render, screen } from 'utils/test-utils'

import Book from '.'

const props = {
  id: '1',
  title: 'HOOKED',
  author: 'Nir Eyal',
  image: 'img/Book.png'
}

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const push = jest.fn()

useRouter.mockImplementation(() => ({
  push,
  query: '',
  asPath: '',
  route: '/'
}))

describe('<Book >', () => {
  it('should render correctly', async () => {
    const { container } = render(<Book {...props} />)

    expect(await screen.findByText(/HOOKED/i)).toBeInTheDocument()
    expect(await screen.findByText(/Nir Eyal/i)).toBeInTheDocument()
    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.image
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render default image correctly', async () => {
    render(<Book author={props.author} id={props.id} title={props.title} />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      'img/cover.png'
    )
    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'alt',
      props.title
    )
  })

  it('should redirect to details page', () => {
    render(<Book {...props} />)

    userEvent.click(screen.getByTestId(props.id))
    expect(push).toHaveBeenCalledWith(`details/${props.id}`)
  })
})
