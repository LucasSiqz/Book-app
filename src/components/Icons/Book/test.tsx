import { render } from 'utils/test-utils'

import Book from '.'

describe('<BookIcon />', () => {
  it('should render correctly', () => {
    const { container } = render(<Book />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
