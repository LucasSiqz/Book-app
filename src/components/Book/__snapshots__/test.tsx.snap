// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book > should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-transition: opacity 0.3s ease-in-out;
  transition: opacity 0.3s ease-in-out;
}

.c0:hover {
  opacity: 0.8;
}

.c0 strong {
  color: rgba(49,49,49,0.8);
  font-size: 1.2rem;
}

.c0 strong:nth-child(3) {
  font-size: 1rem;
  margin-top: 0.5rem;
}

.c1 {
  border-radius: 0.4rem;
  width: 10rem;
  height: 15rem;
  -webkit-filter: drop-shadow(0px 2px 4px rgba(229,229,229,0.5));
  filter: drop-shadow(0px 2px 4px rgba(229,229,229,0.5));
  margin-bottom: 1rem;
}

<div
  class="c0"
  data-testid="1"
>
  <img
    alt="HOOKED"
    class="c1"
    src="img/Book.png"
  />
  <strong>
    HOOKED
  </strong>
  <strong>
    by Nir Eyal
  </strong>
</div>
`
